app/src/main/java/com/odysee/app/utils/Lighthouse.java

```diff
@@ -38,54 +42,58 @@ private static Map<String, Object> buildSearchOptionsKey(String rawQuery, int si
-        Request request = new Request.Builder().url(uriBuilder.toString()).build();
-        OkHttpClient client = new OkHttpClient();
-        ResponseBody responseBody = null;
-        Response response = null;
-        try {
-            response = client.newCall(request).execute();
-            if (response.code() == 200) {
-                responseBody = response.body();
-                if (responseBody != null) {
-                    JSONArray array = new JSONArray(responseBody.string());
-                    for (int i = 0; i < array.length(); i++) {
-                        Claim claim = Claim.fromSearchJSONObject(array.getJSONObject(i));
-                        results.add(claim);
+            List<Claim> results = new ArrayList<>();
+            Request request = new Request.Builder().url(uriBuilder.toString()).build();
+            OkHttpClient client = new OkHttpClient.Builder().readTimeout(30, TimeUnit.SECONDS).build();
+            ResponseBody responseBody = null;
+            Response response = null;
+            try {
+                response = client.newCall(request).execute();
+                if (response.code() == 200) {
+                    responseBody = response.body();
+                    if (responseBody != null) {
+                        JSONArray array = new JSONArray(responseBody.string());
+                        for (int i = 0; i < array.length(); i++) {
+                            Claim claim = Claim.fromSearchJSONObject(array.getJSONObject(i));
+                            results.add(claim);
+                        }
                     }
+                    searchCache.put(cacheKey, results);
+                } else {
+                    throw new LbryResponseException(response.message());
+                }
+            } catch (IOException ex) {
+                throw new LbryRequestException(String.format("search request for '%s' failed", rawQuery), ex);
+            } catch (JSONException ex) {
+                throw new LbryResponseException(String.format("the search response for '%s' could not be parsed", rawQuery), ex);
+            } finally {
+                if (responseBody != null) {
+                    response.close();
                 }
-                searchCache.put(cacheKey, results);
-            } else {
-                throw new LbryResponseException(response.message());
-            }
-        } catch (IOException ex) {
-            throw new LbryRequestException(String.format("search request for '%s' failed", rawQuery), ex);
-        } catch (JSONException ex) {
-            throw new LbryResponseException(String.format("the search response for '%s' could not be parsed", rawQuery), ex);
-        } finally {
-            if (responseBody != null) {
-                response.close();
             }
-        }
 
-        return results;
+            return results;
+        } catch (UnsupportedEncodingException e) {
+            throw new RuntimeException(e);
```

Comment:
This probably will never happen, but we shouldn't rethrow the exception which can result in the app crashing. Best to just ignore it.

Reply:
I prefer to print the stack trace to make it easier to debug.